analysis_driver/util/bash_commands.py

```diff
@@ -8,11 +8,12 @@
 app_logger = log_cfg.get_logger(__name__)
 
 
-def bcl2fastq_per_lane(input_dir, fastq_dir, sample_sheet, masks, lanes):
+def bcl2fastq_per_lane(input_dir, fastq_dir, sample_sheet_func, masks, lanes):
+    # Note sample_sheet_func is the function to get the samplesheet not the samplesheet file itself
```

Comment:
This is a bit of a hack. I didn't want to change where the iteration over 8 lanes happened so had to receive a function that generate the samplesheet instead of the samplesheet iteself.